VariantShredding.md

```diff
@@ -25,276 +25,302 @@
 
-If a value is an object, and the `object` field is present but does not contain all of the fields in the value, then any remaining fields are stored in an object in `variant_value`.
-In the Parquet example above, if field `b` was an object of the form `{"c": 1, "d": 2}"`, then the object `{"d": 2}` would be stored in `variant_value`, and the `c` field would be shredded recursively under `object.c`.
+#### Arrays
 
-Note that an array is always fully shredded if there is an `array` field, so the above consideration for `object` is not relevant for arrays: only one of `array` or `variant_value` may be non-null at a given level.
+Arrays can be shredded using a 3-level Parquet list for `typed_value`.
 
-# Using variant_value vs. typed_value
+If the value is not an array, `typed_value` must be null.
+If the value is an array, `value` must be null.
 
-In general, it is desirable to store values in the `typed_value` field rather than the `variant_value` whenever possible.
-This will typically improve encoding efficiency, and allow the use of Parquet statistics to filter at the row group or page level.
-In the best case, the `variant_value` fields are all null and the engine does not need to read them (or it can omit them from the schema on write entirely).
-There are two main motivations for including the `variant_value` column:
+The list `element` must be a required group that contains optional `value` and `typed_value` fields.
+The element's `value` field stores the element as Variant-encoded `binary` when the `typed_value` cannot represent it.
 
-1) In a case where there are rare type mismatches (for example, a numeric field with rare strings like “n/a”), we allow the field to be shredded, which could still be a significant performance benefit compared to fetching and decoding the full value/metadata binary.
-2) Since there is a single schema per file, there would be no easy way to recover from a type mismatch encountered late in a file write. Parquet files can be large, and buffering all file data before starting to write could be expensive. Including a variant column for every field guarantees we can adhere to the requested shredding schema.
+For example, a `tags` Variant may be shredded as a list of strings using the following definition:
+```
+optional group tags (VARIANT) {
+  required binary metadata;
+  optional binary value;
+  optional group typed_value (LIST) {   # must be optional to allow a null list
+    repeated group list {
+      required group element {          # shredded element
+        optional binary value;
+        optional binary typed_value (STRING);
+      }
+    }
+  }
+}
+```
 
-# Data Skipping
+All elements of an array must be non-null because `array` elements in a Variant cannot be missing.
+That is, either `typed_value` or `value` (but not both) must be non-null.
+Null elements must be encoded in `value` as Variant null: basic type 0 (primitive) and physical type 0 (null).
```

Comment:
just for consistency it was written as 

```
 `00` (Variant null).
 ```
 
 Earlier in the doc but this is fine too